<?xml version="1.0"?>
<launch>  
  <!-- Parameters to pass to the ROS node -->
  <arg name="joint_position_topic" default="/iiwa/state/JointPosition"/>
  <arg name="cartesian_position_topic" default="/iiwa/state/CartesianPose"/>
  <arg name="command_cartesian_position_topic" default="/iiwa/command/CartesianPose"/>
  <arg name="command_joint_position_topic" default="/iiwa/command/JointPosition"/>
  <arg name="use_cartesian_command" default="true"/>
  <arg name="ros_rate" default="0.1"/>
  
  <rosparam command="load" file="$(find iiwa_stack_motion_test)/cfg/iiwa_stack_motion_test_configparam.yaml"/>
  
  <param name="/iiwa/publishJointStates" type="boolean" value="true" /> <!--publish joint states (avoid joint_state_controller)-->
  
  <!--  This loads the robot description  -->
  <include ns="iiwa" file="$(find iiwa_moveit)/launch/planning_context.launch">
    <arg name="load_robot_description" value="true"/>
  </include>
  
  <!-- This publishes the state of the robot   -->
  <node ns="iiwa" name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
	respawn="false" output="screen" />
  
  <!-- Here we call the ROS node we have written, with the parameters on top   -->
  <node ns="iiwa" name="iiwastackMotionTest" pkg="iiwa_stack_motion_test" type="iiwa_stack_motion_test_node" respawn="false" output="screen">
    <param name="joint_position_topic" value="$(arg joint_position_topic)"/>
    <param name="cartesian_position_topic" value="$(arg cartesian_position_topic)"/>
    <param name="command_cartesian_position_topic" value="$(arg command_cartesian_position_topic)"/>
    <param name="command_joint_position_topic" value="$(arg command_joint_position_topic)"/>
    <param name="use_cartesian_command" value="$(arg use_cartesian_command)"/>
    <param name="ros_rate" value="$(arg ros_rate)"/>
  </node>
  
  <!-- Load RViZ with the given configuration file  -->
  <!--node ns="iiwa" name="$(anon rviz)" pkg="rviz" type="rviz" respawn="true"  args="-d $(find iiwa_notool_moveit)/launch/moveit.rviz" output="screen" /-->
  
</launch>

